Update math::Transpose to support tensor with size > 2G

xiaomengy opened the pull request:
Summary:
Update math::Transpose to support tensor with size > 2G

i-am-not-moving-c2-to-c10

Differential Revision: D14313624

houseroad commented:
@pytorchbot rebase this please

houseroad commented:
The windows build is broken. https://ci.pytorch.org/jenkins/job/pytorch-builds/job/pytorch-win-ws2016-cuda9-cudnn7-py3-build/42613/console

xiaomengy committed:
Update math::Transpose to support tensor with size > 2G (#17670)

Summary:
Pull Request resolved: https://github.com/pytorch/pytorch/pull/17670

Update math::Transpose to support tensor with size > 2G

i-am-not-moving-c2-to-c10

Differential Revision: D14313624

fbshipit-source-id: 58cad66e7c05d6f38ba297bb993b456e03a3de70
